src/internal.h

```diff
@@ -154,6 +154,8 @@
 #define DISPATCH_PURE_C 1
 #endif
 
+#include <pthread.h>
```

Comment:
@MadCoder `pthread_attr_t` is required by `private/queue_private.h`. In other platforms, this type gets declared by an include in `sys/types.h` to `pthreadtypes.h`.

Android says in `sys/types.h:106`

```
/* while POSIX wants these in <sys/types.h>, we
 * declare then in <pthread.h> instead */
#if 0
typedef  .... pthread_attr_t;
typedef  .... pthread_cond_t;
typedef  .... pthread_condattr_t;
typedef  .... pthread_key_t;
typedef  .... pthread_mutex_t;
typedef  .... pthread_once_t;
typedef  .... pthread_rwlock_t;
typedef  .... pthread_rwlock_attr_t;
typedef  .... pthread_t;
#endif
```

so we need to include it before `private/queue_private.h`. I can put it in that file instead if preferred.